### PR TITLE
New Timesaver option : Fast Shadow Boat

### LINKED ISSUE #922: Speed up Master Sword cutscenes and Shadow boat
Is it possible to remove/shorten the mastersword cutscenes? 

Also, can we remove the Shadow Temple boat cutscene?

Is there a reason why the song of storms item check requires playing the ocarina, when most of the other song item checks don't require it?

### PR DESCRIPTION
**Pull request by GSKirox**
This new option massively speeds up the boat ride sequence from around 1 min to 10 sec.
The cutscene at the start has been changed to both : 
* Better fit the new behaviour by changing up the starting camera view and cutting the length short.
* Last long enough to prevent jumping directly to the skulltula on the left, which would break logic.

https://github.com/OoTRandomizer/OoT-Randomizer/assets/65768236/3fdb3acd-f6e8-452d-ae69-1ffa2b86e6d6

### CODE REVIEW

File: Cutscenes.py [hunk 1]
```diff
@@ -69,7 +70,7 @@ def patch_cutscene_misc_command(rom: Rom, address: int, start_frame:int, end_fra
     rom.write_int16(address + 2, start_frame)
     rom.write_int16(address + 4, end_frame)
 
-def patch_cutscenes(rom: Rom, songs_as_items:bool) -> None:
+def patch_cutscenes(rom: Rom, songs_as_items:bool, settings:Settings) -> None:
```
Comment (fenhl) [inline Cutscenes.py]:
```suggestion
def patch_cutscenes(rom: Rom, songs_as_items: bool, settings: Settings) -> None:
```

File: SettingsList.py [hunk 2]
```diff
@@ -3277,6 +3277,15 @@ class SettingInfos:
         shared         = True,
     )
 
+    fast_shadow_boat = Checkbutton(
+        gui_text       = 'Fast Shadow Boat',
+        gui_tooltip    = '''\
+            The boat sequence in Shadow Temple will be massively sped up.
```
Comment (fenhl) [inline SettingsList.py:3283]: Maybe the description should mention how this changes the stalfos fight?

**Review by fenhl — CHANGES_REQUESTED** [review]

File: SettingsList.py (hunk 2 above)
Comment (GSKirox) [reply SettingsList.py:3283]: Added a line on Stalfos.

**Comment by GSKirox** [comment]
Impoved the cutscene viewpoint with help from flagrama ! 

https://github.com/OoTRandomizer/OoT-Randomizer/assets/65768236/312deb2c-e750-42ce-b884-fc016b6d9088

**Review by fenhl — APPROVED** [review]

**Comment by fenhl** [comment]
Would it make sense to enable this setting in the Easy Mode preset?

**Pull request merged**